fastlane/Fastfile

```diff
@@ -1127,10 +1129,16 @@ lane :test_without_building do |options|
 
   inject_buildkite_analytics_environment(xctestrun_path: xctestrun_path) if buildkite_ci?
 
+  # Excluding unit tests from parallel testing
+  parallel_testing_value = options[:name] == "UnitTests" ? false : true
+
   run_tests(
     workspace: 'WooCommerce.xcworkspace',
     scheme: TEST_SCHEME,
     device: options[:device],
+    parallel_testing: parallel_testing_value,
+    concurrent_workers: CONCURRENT_SIMULATORS,
+    max_concurrent_simulators: CONCURRENT_SIMULATORS,
```

Comment:
I tried playing around with this and couldn't find any other way to get the tests to run in parallel via `xcodebuild` 😞 

Could be an `xcodebuild` bug in not reading the settings from the test plan.

Reply:
I think the problem is simply that we don't provide the `testplan` parameter to the call to `run_tests`, so that's why `run_tests` and `xcodebuild` don't use it.

We do pass `xctestrun:` parameter to this `run_tests` action, but that's not the same as the `testplan:` parameter, that we should probably pass as well.

Indeed, at first I thought that the `*.xctestrun` file was kind of the "compiled version" of the `*.xctestplan`, and thus that the `*.xctestrun` file would be expected to contain all the info that we set in the `*.xctestplan` file, including that `parallelizable: true` added here. But… after some investigation, I noticed that, while the `*.xctestplan` has `parallelizable: true` attribute stored in its JSON, if you open the `*.xctestrun` (plist file) that is generated in `DerivedData` (after running `build_for_testing`), I don't see any entry about parallelization in the `*.xctestrun` file itself!

So I think our error is that we should always have been passing **both** `xctestrun:` and `testplan:` parameters to `run_tests` (as opposed to only passing `xctestrun:` to it like we apparently have always been doing).

I didn't test the theory but I'm pretty hopeful that if we pass the right `testplan:` parameter to this `run_tests` invocation, then it would not be necessary anymore to pass `parallel_testing: parallel_testing_value` explicitly to the `run_tests` call anymore, and instead `xcodebuild` would finally be able to read that from the test plan.

As for `concurrent_workers` and `max_concurrent_simulators`, this is a setting at the `xcodebuild` level, not a value included in the `*.xctestplan` itself; so I believe that for those we still need to pass them explicitly here when invoking `run_tests` (which invokes `xcodebuild` under the hood).

Reply:
>  And I am guessing you assumed the same too.

Yes, I did. Ah, that explains it, then. I'm keen to explore this 🔍 

> As for concurrent_workers and max_concurrent_simulators, this is a setting at the xcodebuild level, not a value included in the *.xctestplan itself; 

As far as I understand them, they are a way override what would otherwise an unpredictable automatic behavior. Xcode has a similar option in the main page of the preferences, although less granular:

<img width="942" alt="Screenshot 2023-05-01 at 12 28 14 pm" src="https://user-images.githubusercontent.com/1218433/235393021-161a567f-7a00-4189-85be-24df1bdaf8e8.png">

Reply:
> Are these parameters necessary?

as you all have discussed and figured out, this is necessary for now. i've tried a couple of things before landing this one, but none of the attempts work (only turning it on the value from [.xctestplan](https://github.com/woocommerce/woocommerce-ios/pull/9514/files#diff-cdd20a96864e3554162e0b9b303b924f05ebd14e2947a634eb291687be3e5d31) didn't work, adding the commands in xcodebuild on [Rakefile](https://github.com/woocommerce/woocommerce-ios/blob/trunk/Rakefile#L246-L259) without adding the parameters in the fastlane file also didn't work)

if passing the _right_ `testplan` value would simplify this, we can definitely update this later after that's tested.